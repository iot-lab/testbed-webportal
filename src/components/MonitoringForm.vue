<template>
  <form @submit.prevent="updateProfile" class="mt-3">
    <div class="form-group">
      <label class="mr-3 my-label">Name</label>
      <input v-model="profile.profilename" type="text" class="form-control" placeholder="Profile name" ref="name"
        style="max-width: 300px; display: inline-block"
        :class="{'is-invalid': !nameValidation}">
      <div class="invalid-feedback">
        Invalid name. Only alphanumeric characters allowed [0-9A-Za-z_]
      </div>
    </div>
    <div class="form-group">
      <label class="mr-3 my-label">Architecture</label>
      <label class="custom-control custom-radio">
        <input name="radio-archi" type="radio" class="custom-control-input" v-model="profile.nodearch" value="m3">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">M3</span>
      </label>
      <label class="custom-control custom-radio">
        <input name="radio-archi" type="radio" class="custom-control-input" v-model="profile.nodearch" value="a8">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">A8</span>
      </label>
      <label class="custom-control custom-radio">
        <input name="radio-archi" type="radio" class="custom-control-input" v-model="profile.nodearch" value="custom">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Other</span>
      </label>
    </div>
    <div class="form-group">
      <label class="mr-3 my-label">Monitor</label>
      <label class="custom-control custom-checkbox">
        <input name="checkbox-monitor" type="checkbox" class="custom-control-input" v-model="showConso">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Consumption <i class="fa fa-bolt text-muted"></i></span>
      </label>
      <label class="custom-control custom-checkbox">
        <input name="checkbox-monitor" type="checkbox" class="custom-control-input" v-model="showRadio">
        <span class="custom-control-indicator"></span>
        <span class="custom-control-description">Radio <i class="fa fa-wifi text-muted"></i></span>
      </label>
    </div>
    <fieldset class="card bg-light pt-2 mb-3" v-if="showConso">
      <div class="form-group">
        <label class="mr-3 my-label"><i class="fa fa-bolt"></i></label>
        <label class="custom-control custom-checkbox">
          <input name="checkbox-conso" type="checkbox" class="custom-control-input" v-model="profile.consumption.current">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description">Current</span>
        </label>
        <label class="custom-control custom-checkbox">
          <input name="checkbox-conso" type="checkbox" class="custom-control-input" v-model="profile.consumption.voltage">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description">Voltage</span>
        </label>
        <label class="custom-control custom-checkbox">
          <input name="checkbox-conso" type="checkbox" class="custom-control-input" v-model="profile.consumption.power">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description">Power</span>
        </label>
      </div>
      <div class="form-group">
        <label class="mr-3 my-label">Period</label>
        <select class="custom-select" v-model="profile.consumption.period">
          <option value="140">140 µs</option>
          <option value="204">204 µs</option>
          <option value="332">332 µs</option>
          <option value="588">588 µs</option>
          <option value="1100">1100 µs</option>
          <option value="2116">2116 µs</option>
          <option value="4156">4156 µs</option>
          <option value="8244">8244 µs</option>
        </select>
      </div>
      <div class="form-group">
        <label class="mr-3 my-label">Average</label>
        <select class="custom-select" v-model="profile.consumption.average" placeholder="avg">
          <option value="1">1</option>
          <option value="4">4</option>
          <option value="16">16</option>
          <option value="64">64</option>
          <option value="256">256</option>
          <option value="128">428</option>
          <option value="512">512</option>
          <option value="1024">1024</option>
        </select>
      </div>
    </fieldset>
    <fieldset class="card bg-light pt-2 mb-3" v-if="showRadio">
      <div class="form-group">
        <label class="mr-3 my-label"><i class="fa fa-wifi"></i></label>
        <label class="custom-control custom-radio">
          <input name="radio-mode" type="radio" class="custom-control-input" v-model="profile.radio.mode" value="rssi">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description">rssi</span>
        </label>
        <label class="custom-control custom-radio">
          <input name="radio-mode" type="radio" class="custom-control-input" v-model="profile.radio.mode" value="sniffer">
          <span class="custom-control-indicator"></span>
          <span class="custom-control-description">sniffer</span>
        </label>
      </div>
      <div class="form-group">
        <label class="mr-3 my-label">Channels</label>
        <multiselect v-model="profile.radio.channels"
          :placeholder="`select ${profile.radio.mode === 'rssi' ? 'channels' : 'channel'}`"
          :options="[11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]"
          :allow-empty="true"
          :searchable="true"
          :show-labels="false"
          :close-on-select="profile.radio.mode === 'sniffer'"
          :multiple="profile.radio.mode === 'rssi'"
          :class="{'mymultiselect': true, 'invalid': profile.radio.channels.length === 0}"
          style="max-width: 300px; display: inline-block; z-index: 3">
        </multiselect>
        <!-- <div class="invalid-feedback" :style="{'display': (dirty.category && !profile.category) ? 'block': 'none'}">
          The user category field is required.
        </div> -->
      </div>
      <div class="form-group" v-if="profile.radio.mode === 'rssi'">
        <label class="mr-3 my-label">Period</label>
        <div class="input-group" style="display: inline-flex; max-width: 300px;">
          <input v-model="profile.radio.period" type="number" class="form-control" min="1" max="65535" step="1">
          <span class="input-group-addon">ms</span>
        </div>
      </div>
      <div class="form-group" v-if="profile.radio.mode === 'rssi' && profile.radio.channels && profile.radio.channels.length > 1">
        <label class="mr-3 my-label">Measures</label>
        <div class="input-group" style="display: inline-flex; max-width: 300px;">
          <input v-model="profile.radio.num_per_channel" type="range" max="255" min="1" step="1">
          <span class="xinput-group-addon">{{profile.radio.num_per_channel}} per channel</span>
        </div>
      </div>
    </fieldset>
    <div class="form-group">
      <!-- <button class="btn" type="submit">Back</button> -->
      <button class="btn btn-success" type="submit">Save profile</button>
    </div>

    <!-- <div class="form-group">
      <label class="form-control-label">Architecture</label>
      <input placeholder="Profile name" v-model="profile.profilename" name="profilename"
        class="form-control" type="text" v-validate="'required'"
        :class="{'is-invalid': errors.has('profilename') }">
      <div class="invalid-feedback" v-show="errors.has('profilename')">
        {{ errors.first('profilename') }}
      </div>
    </div> -->

  </form>
</template>

<script>
import Multiselect from 'vue-multiselect'
import { iotlab } from '@/rest'
// import { Validator } from 'vee-validate'

export default {
  name: 'MonitoringForm',
  components: { Multiselect },

  props: {
    monitoringProfile: {
      type: Object,
      default: () => {},
    },
  },

  data () {
    return {
      showConso: false,
      showRadio: false,
      profile: {
        profilename: '',
        power: 'dc',
        consumption: {
          current: false,
          voltage: false,
          power: false,
          period: 8244,
          average: 4,
        },
        radio: {
          mode: 'rssi',
          channels: [11],
          period: 1000,
          num_per_channel: 1,
        },
      },
    }
  },

  mounted () {
    if (!this.monitoringProfile.profilename) {
      this.$refs.name.focus()
    }
  },

  watch: {
    monitoringProfile: function (profile) {
      this.profile = Object.assign({}, this.profile, profile)
      this.showConso = this.profileHasConsumption(profile)
      this.showRadio = profile.radio !== undefined
      this.$refs.name.blur()
    },
  },

  computed: {
    nameValidation () {
      return this.profile.profilename.match(/^[0-9A-Za-z_]*$/)
    },
  },

  methods: {
    async validate () {
      // return this.$validator.validateAll().then((validated) => {
      //   if (!validated || !this.profile.country || !this.profile.category) {
      //     this.dirty.category = this.dirty.country = true
      //     console.log('User form is not valid.')
      //     return false
      //   }
      //   return true
      // })
    },

    profileHasConsumption (profile) {
      if (!profile.consumption) return false
      return profile.consumption.current || profile.consumption.voltage || profile.consumption.power
    },
    async updateProfile () {
      let profile = Object.assign({}, this.profile)
      if (profile.radio.mode === 'sniffer') {
        delete profile.radio.period
        delete profile.radio.num_per_channel
        if (typeof profile.radio.channels === 'object') profile.radio.channels = [profile.radio.channels[0]]
        if (typeof profile.radio.channels === 'number') profile.radio.channels = [profile.radio.channels]
      } else {
        if (profile.radio.channels.length < 2) profile.radio.num_per_channel = 0
      }
      if (!this.showConso) delete profile.consumption
      if (!this.showRadio) delete profile.radio

      if (!profile.profilename) {
        this.$notify({text: 'Name is empty', type: 'warning'})
        this.$refs.name.focus()
        return
      }
      if (!profile.nodearch) {
        this.$notify({text: 'Select Architecture first', type: 'warning'})
        return
      }
      if (!this.profileHasConsumption(profile) && !this.showRadio) {
        // this.$notify({text: `${JSON.stringify(profile)}`, type: 'info', duration: 10000})
        this.$notify({text: 'Select something to monitor', type: 'warning'})
        console.log(profile)
        return
      }

      try {
        if (this.monitoringProfile.profilename) {
          // modify existing profile
          await iotlab.updateMonitoringProfile(this.monitoringProfile.profilename, profile)
        } else {
          // create new profile
          await iotlab.createMonitoringProfile(profile)
        }
        this.$notify({text: `Profile ${profile.profilename} saved`, type: 'success'})
        this.$router.push({name: 'listMonitoring'})
      } catch (err) {
        this.$notify({text: err.response.data.message, type: 'error'})
        // this.$notify({text: `${JSON.stringify(profile)}`, type: 'info', duration: 10000})
        console.log(profile)
      }
    },
  },
}
</script>

<style scoped>

</style>
