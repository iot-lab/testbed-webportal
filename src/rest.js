import axios from 'axios'

const API_URL = `https://${process.env.VUE_APP_IOTLAB_HOST}/api/`

export const iotlab = {

  api: axios.create({baseURL: API_URL, auth: JSON.parse(localStorage.getItem('apiAuth') || '{}')}),

  create (username, password) {
    this.api = axios.create({
      baseURL: API_URL,
      auth: {
        username: username,
        password: password,
      },
    })
  },

  // USERS API

  async signup (user, nbUsers) {
    if (nbUsers) {
      // create multiple user accounts
      await iotlab.api.post(`/users?nbusers=${nbUsers}`, user)
    } else {
      // create single user account
      await iotlab.api.post('/users', user)
    }
  },

  async activateAccount (hash) {
    await iotlab.api.post('/users/activate/', {'hash': hash})
  },

  async deleteUser (login = undefined) {
    if (login) {
      await iotlab.api.delete(`/users/${login}`)
    } else {
      await iotlab.api.delete(`/user`)
    }
  },

  async resetPassword (email) {
    await iotlab.api.post(`/users/reset_password`, {'email': email})
  },

  async changePassword (oldPwd, newPwd, confirmPwd) {
    await iotlab.api.put('/user/password', {
      'old_password': oldPwd,
      'new_password': newPwd,
      'confirm_new_password': confirmPwd,
    })
  },

  async getSSHkeys () {
    return iotlab.api.get('/user/keys').then(resp => resp.data.sshkeys)
  },

  async modifySSHkeys (keys) {
    return iotlab.api.post('/user/keys', {'sshkeys': keys})
  },

  async deleteSSHkey (id) {
    return iotlab.api.delete(`/user/keys/${id}`)
  },

  async getUsers ({status, group, search} = {}) {
    let params = {}
    if (status) {
      params.status = status
    }
    if (group) {
      params.group = group
    }
    if (search) {
      params.search = search
    }
    return iotlab.api.get('/users', {params: params}).then(resp => resp.data)
  },

  async getUserInfo (login = undefined) {
    if (login) {
      return iotlab.api.get(`/users/${login}`).then(resp => resp.data)
    } else {
      return iotlab.api.get('/user').then(resp => resp.data)
    }
  },

  async setUserInfo (user, login = undefined) {
    if (login) {
      return iotlab.api.put(`/users/${login}`, user)
    } else {
      return iotlab.api.put('/user', user)
    }
  },

  async activateUser (login) {
    return iotlab.api.put(`/users/${login}`, {'status': 'active'})
  },

  async deactivateUser (login) {
    return iotlab.api.put(`/users/${login}`, {'status': 'inactive'})
  },

  async setUserGroups (login, groups) {
    return iotlab.api.put(`/users/${login}`, {'groups': groups})
  },

  async sendMail (recipients, subject, message) {
    await iotlab.api.post('/users/email', {
      recipients: recipients,
      subject: subject,
      body: message,
    })
  },

  // EXPERIMENT API

  async getUserExperimentsCount () {
    return iotlab.api.get('/experiments/total').then(resp => resp.data)
  },

  async getAllExperimentsCount (user) {
    if (user) {
      return iotlab.api.get(`/experiments/total/users/${user}`).then(resp => resp.data)
    } else {
      return iotlab.api.get(`/experiments/total/all`).then(resp => resp.data)
    }
  },

  async getAllExperiments ({limit, offset, user, state = 'Terminated,Error,Running,Finishing,Resuming,toError,Waiting,Launching,Hold,toLaunch,toAckReservation,Suspended'} = {}) {
    let params = {}
    if (state) {
      params.state = state
    }
    if (limit) {
      params.limit = Math.min(limit, 500) // IoT-LAB api restrict queries to 500 entries
    }
    if (offset) {
      params.offset = offset
    }
    if (user === '@self') {
      return iotlab.api.get('/experiments', {params: params}).then(resp => resp.data.items)
    } else if (user) {
      return iotlab.api.get(`/experiments/users/${user}`, {params: params}).then(resp => resp.data.items)
    } else {
      return iotlab.api.get('/experiments/all', {params: params}).then(resp => resp.data.items)
    }
  },

  async getRunningExperiments () {
    return iotlab.api.get('/experiments/running').then(resp => resp.data.items)
  },

  async getExperiment (id) {
    return iotlab.api.get(`/experiments/${id}`).then(resp => resp.data)
  },

  async getExperimentArchive (id) {
    return iotlab.api.get(`/experiments/${id}/data`, {responseType: 'arraybuffer'}).then(resp => resp.data)
  },

  async getExperimentDeployment (id) {
    return iotlab.api.get(`/experiments/${id}/deployment`).then(resp => resp.data)
  },

  async stopExperiment (id) {
    return iotlab.api.delete(`/experiments/${id}`)
  },

  async submitExperiment ({type, name, duration, reservation, nodes, firmwareassociations, firmwares, profileassociations, profiles} = {}) {
    const formData = new FormData()
    formData.append('experiment.json', JSON.stringify({
      type: type,
      duration: duration,
      reservation: reservation,
      name: name,
      nodes: nodes,
      firmwareassociations: firmwareassociations,
      profileassociations: profileassociations,
    }))
    // attached firmware names should be unique
    let attachedFirmwares = []
    for (let asso of firmwareassociations || []) {
      if (asso.firmwarename in firmwares && !attachedFirmwares.includes(asso.firmwarename)) {
        formData.append(asso.firmwarename, new Blob([firmwares[asso.firmwarename]], {type: 'application/octet-stream'}), asso.firmwarename)
        attachedFirmwares.push(asso.firmwarename)
      }
    }
    return iotlab.api.post('/experiments', formData).then(resp => resp.data)
  },

  async checkFirmware (fileString) {
    const formData = new FormData()
    formData.append('firmware.bin', fileString)
    return iotlab.api.post('/firmwares/checker', formData).then(resp => resp.data)
  },

  async flashFirmware (id, nodes, fileString) {
    const formData = new FormData()
    formData.append(`experiment${id}.json`, JSON.stringify(nodes))
    formData.append('firmware.bin', new Blob([fileString], {type: 'application/octet-stream'}), 'firmware.bin')
    return iotlab.api.post(`/experiments/${id}/nodes/flash`, formData).then(resp => resp.data)
  },

  async flashResourcesFirmware (id, nodes, name) {
    return iotlab.api.post(`/experiments/${id}/nodes/flash/${name}`, nodes).then(resp => resp.data)
  },

  async updateMonitoring (id, nodes, name) {
    return iotlab.api.post(`/experiments/${id}/nodes/monitoring/${name}`, nodes).then(resp => resp.data)
  },

  async updateExperimentMobilityCircuit (id, nodes, name) {
    return iotlab.api.post(`/experiments/${id}/robots/mobility/${name}`, nodes).then(resp => resp.data)
  },

  async reloadExperiment (id) {
    return iotlab.api.post(`/experiments/${id}/reload`, {}).then(resp => resp.data)
  },

  // USER GROUPS

  async getUserGroups () {
    return iotlab.api.get('/groups').then(resp => resp.data)
  },

  async createGroup (name) {
    return iotlab.api.post(`/groups`, {name: name})
  },

  async deleteGroup (group) {
    return iotlab.api.delete(`/groups/${group}`)
  },

  // OTHER API

  async getSites () {
    return iotlab.api.get('/sites').then(resp => resp.data.items)
  },

  async getSitesDetails () {
    return iotlab.api.get('/sites/details').then(resp => resp.data.items)
  },

  async getNodes () {
    return iotlab.api.get('/nodes').then(resp => resp.data.items)
  },

  async updateNodesProperties () {
    return iotlab.api.put('/nodes')
  },

  async updateNodesPropertiesOldApi () {
    const OLD_API = `https://${process.env.VUE_APP_IOTLAB_HOST}/rest/`
    const oldApi = axios.create({baseURL: OLD_API, auth: JSON.parse(localStorage.getItem('apiAuth') || '{}')})
    return oldApi.post('/admin/nodes')
  },

  async getNodesIds () {
    return iotlab.api.get('/nodes/ids').then(resp => resp.data.items)
  },

  async getExperimentNodes (id) {
    return iotlab.api.get(`/experiments/${id}/nodes`).then(resp => resp.data.items)
  },

  async getExperimentToken (id) {
    return iotlab.api.get(`/experiments/${id}/token`).then(resp => resp.data.token)
  },

  async sendNodesCommand (id, cmd, nodes) {
    // cmd in [start, stop, reset, flash-idle, profile-reset, debug-start, debug-stop]
    return iotlab.api.post(`/experiments/${id}/nodes/${cmd}`, nodes).then(resp => resp.data)
  },

  // STORES (firmwares, profiles, etc)

  async getMobilityCircuits () {
    return iotlab.api.get('/mobilities/circuits').then(resp => resp.data.items)
  },

  async getMobilityCircuit (name) {
    return iotlab.api.get(`/mobilities/circuits/${name}`).then(resp => resp.data)
  },

  async getRobotsSiteMapImage (site) {
    return iotlab.api.get(`/robots/${site}/map/image`, {
      responseType: 'arraybuffer',
      headers: {
        'Access-Control-Allow-Origin': '*',
      },
    }).then(resp => Buffer.from(resp.data, 'binary').toString('base64')
    )
  },

  async getRobotsSiteMapConfig (site) {
    return iotlab.api.get(`/robots/${site}/map/config`).then(resp => resp.data)
  },

  async createMobilityCircuit (circuit) {
    return iotlab.api.post(`/mobilities/circuits`, circuit)
  },

  async updateMobilityCircuit (name, circuit) {
    return iotlab.api.put(`/mobilities/circuits/${name}`, circuit)
  },

  async deleteMobilityCircuit (name) {
    return iotlab.api.delete(`/mobilities/circuits/${name}`)
  },

  async getMonitoringProfiles () {
    return iotlab.api.get('/monitoring').then(resp => resp.data)
  },

  async getMonitoringProfile (name) {
    return iotlab.api.get(`/monitoring/${name}`).then(resp => resp.data)
  },

  async updateMonitoringProfile (name, profile) {
    return iotlab.api.put(`/monitoring/${name}`, profile)
  },

  async createMonitoringProfile (profile) {
    return iotlab.api.post(`/monitoring`, profile)
  },

  async deleteMonitoringProfile (name) {
    return iotlab.api.delete(`/monitoring/${name}`)
  },

  async getFirmwares () {
    return iotlab.api.get('/firmwares').then(resp => resp.data)
  },

  async getFirmware (name) {
    return iotlab.api.get(`/firmwares/${name}`).then(resp => resp.data)
  },

  async getFirmwareFile (name) {
    return iotlab.api.get(`/firmwares/${name}/file`, {responseType: 'arraybuffer'}).then(resp => resp.data)
  },

  async updateFirmware (name, firmware, fileString) {
    const formData = new FormData()
    formData.append(`firmware.json`, JSON.stringify(firmware))
    if (fileString) {
      formData.append('firmware.bin', new Blob([fileString], {type: 'application/octet-stream'}), 'firmware.bin')
    }
    return iotlab.api.put(`/firmwares/${name}`, formData).then(resp => resp.data)
  },

  async createFirmware (firmware, fileString) {
    const formData = new FormData()
    formData.append(`firmware.json`, JSON.stringify(firmware))
    formData.append('firmware.bin', new Blob([fileString], {type: 'application/octet-stream'}), 'firmware.bin')
    return iotlab.api.post(`/firmwares`, formData).then(resp => resp.data)
  },

  async deleteFirmware (name) {
    return iotlab.api.delete(`/firmwares/${name}`)
  },
}
