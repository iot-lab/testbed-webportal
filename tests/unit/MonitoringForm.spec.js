import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Notifications from 'vue-notification'
import MonitoringForm from '@/components/MonitoringForm'
import { iotlab } from '@/rest'
jest.mock('@/rest')
iotlab.updateMonitoringProfile = jest.fn()
iotlab.createMonitoringProfile = jest.fn()

const localVue = createLocalVue()
const router = new VueRouter()
localVue.use(VueRouter)
localVue.use(Notifications)
const profiles = [{"profilename":"","consumption":{"average":4,"period":8244,"current":true,"power":false,"voltage":false},"power":"dc","nodearch":"a8"},{"profilename":"conso","consumption":{"average":4,"period":8244,"current":true,"power":true,"voltage":true},"power":"dc","nodearch":"m3"},{"profilename":"radio","power":"dc","nodearch":"custom","radio":{"mode":"rssi","period":1000,"channels":[11,13],"num_per_channel":16}}]

describe('MonitoringForm.vue', () => {

  it('should match snapshot', () => {
    const wrapper = shallowMount(MonitoringForm, {localVue, router, propsData: {
      monitoringProfile: profiles[0]
    }})
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should match snapshot 2', () => {
    const wrapper = shallowMount(MonitoringForm, {localVue, router, propsData: {
      monitoringProfile: profiles[1]
    }})
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should match snapshot 3', () => {
    const wrapper = shallowMount(MonitoringForm, {localVue, router, propsData: {
      monitoringProfile: profiles[2]
    }})
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should update profile and redirect', () => {
    const wrapper = shallowMount(MonitoringForm, {localVue, router, propsData: {
      monitoringProfile: profiles[1]
    }})
    // change monitoring profile for watch() to take effect
    wrapper.setProps({
      monitoringProfile: profiles[2]
    })

    wrapper.find('form').trigger('submit')
    expect(iotlab.updateMonitoringProfile).toHaveBeenCalled()
    
    wrapper.vm.$router.push = jest.fn()
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$router.push).toHaveBeenCalledWith({name: 'resources'})
    })
  })

})
