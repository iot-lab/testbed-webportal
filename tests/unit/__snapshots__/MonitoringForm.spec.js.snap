// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitoringForm.vue should match snapshot 1`] = `
<form
  class="mt-3"
>
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Name
    </label>
     
    <input
      class="form-control"
      placeholder="Profile name"
      style="max-width: 300px; display: inline-block;"
      type="text"
    />
     
    <div
      class="invalid-feedback"
    >
      
      Invalid name. Only alphanumeric characters allowed [0-9A-Za-z_]
    
    </div>
  </div>
   
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Architecture
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="m3"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        M3
      </span>
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="a8"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        A8
      </span>
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="custom"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Other
      </span>
    </label>
  </div>
   
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Monitor
    </label>
     
    <label
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        name="checkbox-monitor"
        type="checkbox"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Consumption 
        <i
          class="fa fa-bolt text-muted"
        />
      </span>
    </label>
     
    <label
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        name="checkbox-monitor"
        type="checkbox"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Radio 
        <i
          class="fa fa-wifi text-muted"
        />
      </span>
    </label>
  </div>
   
  <!---->
   
  <!---->
   
  <div
    class="form-group"
  >
    <button
      class="btn btn-success"
      type="submit"
    >
      Save profile
    </button>
  </div>
</form>
`;

exports[`MonitoringForm.vue should match snapshot 2 1`] = `
<form
  class="mt-3"
>
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Name
    </label>
     
    <input
      class="form-control"
      placeholder="Profile name"
      style="max-width: 300px; display: inline-block;"
      type="text"
    />
     
    <div
      class="invalid-feedback"
    >
      
      Invalid name. Only alphanumeric characters allowed [0-9A-Za-z_]
    
    </div>
  </div>
   
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Architecture
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="m3"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        M3
      </span>
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="a8"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        A8
      </span>
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="custom"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Other
      </span>
    </label>
  </div>
   
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Monitor
    </label>
     
    <label
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        name="checkbox-monitor"
        type="checkbox"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Consumption 
        <i
          class="fa fa-bolt text-muted"
        />
      </span>
    </label>
     
    <label
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        name="checkbox-monitor"
        type="checkbox"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Radio 
        <i
          class="fa fa-wifi text-muted"
        />
      </span>
    </label>
  </div>
   
  <!---->
   
  <!---->
   
  <div
    class="form-group"
  >
    <button
      class="btn btn-success"
      type="submit"
    >
      Save profile
    </button>
  </div>
</form>
`;

exports[`MonitoringForm.vue should match snapshot 3 1`] = `
<form
  class="mt-3"
>
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Name
    </label>
     
    <input
      class="form-control"
      placeholder="Profile name"
      style="max-width: 300px; display: inline-block;"
      type="text"
    />
     
    <div
      class="invalid-feedback"
    >
      
      Invalid name. Only alphanumeric characters allowed [0-9A-Za-z_]
    
    </div>
  </div>
   
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Architecture
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="m3"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        M3
      </span>
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="a8"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        A8
      </span>
    </label>
     
    <label
      class="custom-control custom-radio"
    >
      <input
        class="custom-control-input"
        name="radio-archi"
        type="radio"
        value="custom"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Other
      </span>
    </label>
  </div>
   
  <div
    class="form-group"
  >
    <label
      class="mr-3 my-label"
    >
      Monitor
    </label>
     
    <label
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        name="checkbox-monitor"
        type="checkbox"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Consumption 
        <i
          class="fa fa-bolt text-muted"
        />
      </span>
    </label>
     
    <label
      class="custom-control custom-checkbox"
    >
      <input
        class="custom-control-input"
        name="checkbox-monitor"
        type="checkbox"
      />
       
      <span
        class="custom-control-indicator"
      />
       
      <span
        class="custom-control-description"
      >
        Radio 
        <i
          class="fa fa-wifi text-muted"
        />
      </span>
    </label>
  </div>
   
  <!---->
   
  <!---->
   
  <div
    class="form-group"
  >
    <button
      class="btn btn-success"
      type="submit"
    >
      Save profile
    </button>
  </div>
</form>
`;
